src/main/java/alfio/manager/OpenIdAuthenticationManager.java

```diff
@@ -0,0 +1,178 @@
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.client.utils.URIBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.*;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static alfio.util.HttpUtils.APPLICATION_FORM_URLENCODED;
+import static alfio.util.HttpUtils.APPLICATION_JSON;
+
+@Component
+@Profile("openid")
+public class OpenIdAuthenticationManager
+{
+    private final Logger logger = LoggerFactory.getLogger(OpenIdAuthenticationManager.class);
+
+    private final String domain;
+    private final String clientId;
+    private final String clientSecret;
+    private final String callbackURI;
+    private final String authenticationUrl;
+    private final String claimsUrl;
+    private final String contentType;
+    private final String groupsNameParameter;
+    private final String alfioGroupsNameParameter;
+    private final String logoutUrl;
+    private final String logoutRedirectUrl;
+
+    public OpenIdAuthenticationManager(@Value("${openid.domain}") String domain,
+                                       @Value("${openid.clientId}") String clientId,
+                                       @Value("${openid.clientSecret}") String clientSecret,
+                                       @Value("${openid.callbackURI}") String callbackURI,
+                                       @Value("${openid.authenticationUrl}") String authenticationUrl,
+                                       @Value("${openid.claimsUrl}") String claimsUrl,
+                                       @Value("${openid.contentType}") String contentType,
+                                       @Value("${openid.groupsNameParameter}") String groupsNameParameter,
+                                       @Value("${openid.alfioGroupsNameParameter}") String alfioGroupsNameParameter,
+                                       @Value("${openid.logoutUrl}") String logoutUrl,
+                                       @Value("${openid.logoutRedirectUrl}") String logoutRedirectUrl){
+        this.domain = domain;
+        this.clientId = clientId;
+        this.clientSecret = clientSecret;
+        this.callbackURI = callbackURI;
+        this.authenticationUrl = authenticationUrl;
+        this.claimsUrl = claimsUrl;
+        this.contentType = contentType;
+        this.groupsNameParameter = groupsNameParameter;
+        this.alfioGroupsNameParameter = alfioGroupsNameParameter;
+        this.logoutUrl = logoutUrl;
+        this.logoutRedirectUrl = logoutRedirectUrl;
+    }
+
+    public String buildAuthorizeUrl(List<String> scopes){
+        String scopeParameter = scopes.stream().collect(Collectors.joining(" "));
+        System.out.println(scopeParameter);
+
+        URIBuilder builder = new URIBuilder()
```

Comment:
please use the [org.springframework.web.util.UriComponentsBuilder](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/util/UriComponentsBuilder.html) instead. This will allow you to remove the httpclient dependency

Reply:
done